### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by lcko1012**
## Related Tickets
- [#46897](https://edu-redmine.sun-asterisk.vn/issues/46897)

## WHAT (optional)
- Create sign up function
- Activate account via email

## HOW
- Create sign up form, when user create an account, send an email has activation link.
- Create account activation controller to change `activated` attribute of user to `true`

## Evidence (Screenshot or Video)

image
image
image

### CODE REVIEW

**Review by trinhdvt — APPROVED** [review]
ok

**Comment by lcko1012** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      flash.now[:danger] = t ".failure_message"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update; end
+
+  def destroy; end
```
Comment (ghost) [inline app/controllers/users_controller.rb]: mấy action này chưa dùng đến thì ko nên thêm vào

File: config/routes.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 Rails.application.routes.draw do
   scope "(:locale)", locale: /en|vi/ do
     root "static_pages#home"
+    get "/signup", to: "users#new"
+    resources :users
```
Comment (ghost) [inline config/routes.rb]: dùng những action nào thì e thêm vào thôi nhé

**Review by ghost — COMMENTED** [review]
cmt

**Pull request merged**